pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -66,6 +65,14 @@ type podNIC interface {
 	PlugPhase2(vmi *v1.VirtualMachineInstance, iface *v1.Interface, network *v1.Network, domain *api.Domain, podInterfaceName string) error
 }
 
+type Network struct {
+	podnic podNIC
+}
+
+func New() *Network {
+	return &Network{podnic: &podNICImpl{networkHandler: NewHandler()}}
```

Comment:
I guess this is the core change in this PR.

You structured a Network that has a podNIC member and the podNIC having a Handler.
I'm having hard time to compute this.

Reply:
BTW, why do we need to instantiate a `Network`? I do not see it used in any interesting manner.

Reply:
More or less we have three levels of expose:
- Network struct: expose the functions to the package
- podNIC: expose the Phase1/Phase2 internal functions 
- networkHandler: expose the low level network functions so we can mock them.

Reply:
> BTW, why do we need to instantiate a Network? I do not see it used in any interesting manner.

Is the trick to test the `SetupPodNetwork` functions

```
		podNICFactory = newpodNIC
		networkMock = &Network{podnic: mockpodNIC}
	})

	Context("interface configuration", func() {
		It("should configure bridged pod networking by default", func() {
			podNICFactory = func(network *v1.Network) (podNIC, error) {
				return mockpodNIC, nil
			}
			vm := newVMIBridgeInterface("testnamespace", "testVmName")
			iface := v1.DefaultBridgeNetworkInterface()
			defaultNet := v1.DefaultPodNetwork()

			mockpodNIC.EXPECT().PlugPhase1(vm, iface, defaultNet, primaryPodInterfaceName, pid)
			err := SetupPodNetworkPhase1(vm, pid)
			err := networkMock.SetupPodNetworkPhase1(vm, pid)
			Expect(err).To(BeNil())
		})

```

Reply:
I'm very happy with this PR; I had it as a rough idea and thought of it as a follow up to my PR #4990, but it considered it too difficult for me.

However, I'm not sure about the name and merit of the suggested `Network` type. It has nothing to do with `v1.Network` or any other specific "network" in the code. Frankly, I don't see what it adds on top of the `podNIC` type. If I understand it correctly, it exists only to serve as a receiver for `SetupPodNetworkPhase1`. Would you consider shoving `podNIC` as the receiver, instead? `podNIC` is a badly-named type - but that's a good thing, because it would not confuse the reader to think that it is something else.

Reply:
I think that the `NetworkUtilsHandler` can also be embedded into the `podNICImpl` so the operations will be executed on this strange object and not have it as member.

Reply:
I will try to embedded structs a  little to see if things get simpler also, `NetworkHandler` can be named `NetworkCommands` and we can have the `LinuxNetworkCommands` implementation (at the end is what we have more or less)